syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package postgresGRPC;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc sendPlan (PlannedStmtRPC) returns (HelloReply) {}
  rpc executeOnServer(PlannedStmt) returns (HelloReply) {}

  rpc RunSelect (RunSelectRequest) returns (Response);
  rpc RunInserts (RunInsertsRequest) returns (Response);
}

message StringList {
  repeated string values = 1;
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message PlannedStmtRPC {
  int32 plan_width = 1;
}

message PlannedStmt {
  string planTree = 1;
}

////////////////////////////////////////////////

// Request message for run_select
message RunSelectRequest {
  repeated string column_names = 1;
  int32 search_key = 2;
}

// Response message (can be expanded as needed)
message Response {
  string message = 1;
}

// Request message for run_inserts
message RunInsertsRequest {
  repeated string column_names = 1;
  repeated StringList input_rows = 2;
}